plugins/woocommerce/src/Blocks/BlockTypes/AbstractBlock.php

```diff
@@ -80,6 +87,22 @@ protected function get_full_block_name() {
 		return $this->namespace . '/' . $this->block_name;
 	}
 
+	/**
+	 * Loads the compiled block metadata from a file.
+	 *
+	 * This method ensures the metadata is loaded only once and stored in a static variable.
+	 */
+	protected function load_compiled_block_metadata() {
+		if ( null === self::$compiled_block_metadata ) {
+			$meta_file_path = WC_ABSPATH . '/assets/client/blocks/blocks-json.php';
+			if ( file_exists( $meta_file_path ) ) {
+				self::$compiled_block_metadata = require $meta_file_path;
+			} else {
+				self::$compiled_block_metadata = [];
+			}
+		}
+	}
```

Comment:
I think there's more risk of someone refactoring this in the future and making it non-static and we lose the benefit.

Having it on the assets class also make more sense logically because that's where all shared things between blocks live (like loading scripts, registering them...)